### ISSUE TITLE
How to control order of appearance when placing multiple adjacent fields on pages?

### ISSUE THREAD

**Issue by otherjoel** [hougaard/Simple-Object-Designer#193]
I am adding 4 fields to a page. I want them grouped together on that page and I also want them to appear in a specific order. I can get the first and last items in the group to be placed accurately using AddBefore and AddAfter. But the middle two are tricky because SOD doesn't let me refer to custom fields when selecting anchors for AddBefore and AddAfter.

Is there a reliable trick or deterministic rule I can use to get adjacent custom fields to appear in the correct order when placed on a page? If so, will it still work when publishing the app to another environment (as long as the environments are identical in all other relevant aspects)?

**Comment by otherjoel** [comment]
*bump* … any ideas here?

**Comment by hougaard** [comment]
Sorry, missed you there. Currently, the field numbers are the only way to order the fields. You are allowed to change a field number on a field.

**Comment by otherjoel** [comment]
OK armed with that hint I experimented. The `FieldNo` (in the custom field’s settings) definitely does play a role in determining placement, but it's not everything. 

So in case it helps anyone else, here's how it works.

Say you have three custom fields:

| Field Name       | FieldNo |
|------------------|:-------:|
| Start Date     |  `5001` |
| Middle Date      |  `5002` |
| End Date |  `5003` |

If you add these fields to, say, the Item card, and set all three to `AddAfter` the built-in *Type* field, they will appear in this order:

1. Type
2. End
3. Middle
4. Start

Notice they appear visually in “reverse” order. I would guess this is because SOD somehow process the fields in order: the field whose `FieldNo` is lowest is first inserted directly after the *Type* field, then the next one, then the next one; and after doing the operations in this sequence you naturally end up with the order shown above.

If you want the fields to appear adjacent on the page, from top to bottom in _ascending_ order of their `FieldNo` values, you **must** place all of them using `AddBefore`, and specify the same field for each.

**Comment by hougaard** [comment]
Yeah, not perfect, it's on my list of things to improve... (Together with the option to create new groups).
